File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.entity.PageResult;
import com.pinyougou.pojo.TbBrand;

import java.util.List;

/**
 * 品牌接口
 *
 * @author Administrator
 */
public interface BrandService {

    /**
     * 查询所有数据
     *
     * @return
     */
    List<TbBrand> findAll();

    /**
     * 分页查询
     *
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult findPage(int pageNum, int pageSize);

    /**
     * 添加
     *
     * @param brand
     */
    void addTbBrand(TbBrand brand);

    /**
     * 修改
     *
     * @param brand
     */
    void update(TbBrand brand);

    /**
     * 根据ID获取实体
     *
     * @param id
     * @return
     */
    TbBrand findOne(long id);

    /**
     * 删除
     *
     * @param ids
     */
    void delete(Long[] ids);

    /**
     * 根据条件查询 分页
     *
     * @param brand
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult findPage(TbBrand brand, int pageNum, int pageSize);

}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
/**
 * 品牌服务
 */
app.service("brandService", function ($http) {

    //查询所有
    this.findAll = function () {
        return $http.get("../brand/findAll.do")
    };

    //搜索
    this.search = function (page, size, searchEntity) {
        return $http.post("../brand/search.do?page=" + page + "&size=" + size, searchEntity);
    };

    //分页
    this.findPage = function (page, rows) {
        return $http.get("../brand/findPage.do?page=" + page + "&size=" + rows)
    };

    //删除
    this.dele = function (ids) {
        return $http.get("../brand/delete.do?ids=" + ids)
    };

    //查找某一个
    this.findOne = function (id) {
        return $http.get("../brand/findOne.do?id=" + id);
    };

    //保存or更新
    this.save = function (entity) {
        var url = entity.id != null ? "update" : "add";
        return $http.post("../brand/" + url + ".do", entity);
    };




});